src/Meshes/geometry.ts

```diff
@@ -45,6 +45,7 @@ export class Geometry implements IGetSetVerticesData {
     public onGeometryUpdated: (geometry: Geometry, kind?: string) => void;
 
     // Private
+    private _borrowed_index_buffer: boolean = false;

```

Comment:
I'm afraid that cannot work because what will happen if I dispose the source index buffer?
We need to reference count the index buffers as well if you want to go down this path

Reply:
That makes sense. I thought this through in the VertexBuffer case but index buffer is still hacky!

Reply:
It seems that the index buffer is also a DataBuffer and those are referenceCounted, so I might be able to do something similar. Not sure how the clone-function would look though.

Reply:
Ok tried solving this using references on the index_buffer